audit/audit.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package audit provides an audit log writer for access to secrets.
+package audit
+
+import (
+	"encoding/json"
+	"io"
+	"net/netip"
+	"os"
+
+	"github.com/tailscale/setec/acl"
+	"github.com/tailscale/setec/types/api"
+)
+
+// Principal is the identity of a client taking action on the secrets
+// service.
+type Principal struct {
+	Hostname string     `json:"hostname"`
+	IP       netip.Addr `json:"ip"`
+	User     string     `json:"user,omitempty"`
+	Tags     []string   `json:"tags,omitempty"`
+}
+
+// Entry is an audit log entry.
+type Entry struct {
+	// Principal is the client who is doing something.
+	Principal Principal `json:"principal"`
+	// Action is the action being performed on a secret.
+	Action acl.Action `json:"action"`
+	// Secret is the name of the secret being acted upon.
+	Secret string `json:"secret"`
+	// SecretVersion is the version of the secret being acted upon, if
+	// applicable, or api.SecretVersionDefault if a version doesn't
+	// make sense for the action (e.g. listing a secret).
+	SecretVersion api.SecretVersion `json:"secretVersion,omitempty"`
+}
+
+// Writer is an audit log writer.
+type Writer struct {
+	w io.Writer
+}
+
+// New returns a Writer that outputs audit log entries to w. If w also
```

Comment:
We might want to document that logs are written as JSON, though that is implicit from all the struct tags.

Reply:
Done.